File: all-submissions/1254-deepest-leaves-sum/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    private int findMaxDepth(TreeNode node) {
        if (node == null) return 0;
        return Math.max(findMaxDepth(node.left), findMaxDepth(node.right)) + 1;
    }
    
    public int deepestLeavesSum(TreeNode root) {
        if (root == null) return 0;
        int answer = 0;
        int maxDepth = findMaxDepth(root);
        int currDepth = 1;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while(!queue.isEmpty()) {
            int nodesInCurrLevel = queue.size();
            answer = 0;
            for (int i=0; i< nodesInCurrLevel; i++) {
                TreeNode node = queue.remove();
                if (currDepth == maxDepth) {
                    answer += node.val;
                }
                if (node.left != null){
                    queue.add(node.left);
                }
                if(node.right != null) {
                    queue.add(node.right);
                }
            }
            currDepth++;
        }
        return answer;
        
    }
}

File: all-submissions/1254-deepest-leaves-sum/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        # def maxDepth(node):
        #     if node is None:
        #         return 0
        #     return max(maxDepth(node.left), maxDepth(node.right)) +1
        # max_level = maxDepth(root)
        queue = deque()
        queue.append(root)
        ans = 0
        while queue:
            nodes_len = len(queue)
            ans = 0
            for _ in range (nodes_len):
                node = queue.popleft()
                ans += node.val
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return ans

